client/api/editor/item_events.ts

```diff
@@ -0,0 +1,281 @@
+import {createRef} from 'react';
+
+import {
+    BOOKMARK_TYPE,
+    EDITOR_TYPE,
+    IEditorAPI,
+    IEditorBookmark,
+    IEditorFormGroup,
+    IEventItem,
+    IPlanningItem
+} from '../../interfaces';
+import {planningApi, superdeskApi} from '../../superdeskApi';
+
+import {generateTempId} from '../../utils';
+import {filterProfileForEnabledFields} from '../../utils/forms';
+import {TEMP_ID_PREFIX} from '../../constants';
+
+import {AddPlanningBookmark, AssociatedPlanningsBookmark} from '../../components/Editor/bookmarks';
+import {RelatedPlanningItem} from '../../components/fields/editor/EventRelatedPlannings/RelatedPlanningItem';
+
+
+export function getEventsInstance(type: EDITOR_TYPE): IEditorAPI['item']['events'] {
+    function getGroups() {
+        const {gettext} = superdeskApi.localization;
+        const profile = planningApi.events.getEditorProfile();
+
+        return {
```

Comment:
wondering why not use a list for these, id is duplicating the dict key anyhow and there is extra index

Reply:
This is to make it easier to determine if a bookmark should be rendered, in `getGroupsForItem` function (in the same file):

``` js
groups = getGroups();

return {
    bookmarks: [{
        id: 'location',
        disabled: !groups.location.fields.length,
    }],
    groups: Object.values(groups),
}
```

Reply:
I see